service/src/main/java/co/cask/wrangler/service/directive/DirectivesService.java

```diff
@@ -265,6 +267,106 @@ public void get(HttpServiceRequest request, HttpServiceResponder responder,
   }
 
   /**
+   * Upload data to the workspace, the workspace is created automatically on fly.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   */
+  @POST
+  @Path("workspaces")
+  public void upload(HttpServiceRequest request, HttpServiceResponder responder) {
+
+    try {
+      String name = request.getHeader(PropertyIds.FILE_NAME);
+      String id = ServiceUtils.generateMD5(request.getHeader(name));
+
+      // if workspace doesn't exist, then we create the workspace before
+      // adding data to the workspace.
+      if (!table.hasWorkspace(id)) {
+        table.createWorkspaceMeta(id, id);
```

Comment:
And also perhaps the first column should be `body` instead of the workspace id as well